core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -202,6 +202,17 @@ template <typename T>
 constexpr bool dependent_false_v = !sizeof(T*);
 //==============================================================================
 
+template <typename T>
+T check_mul_of(T a, T b) {
+  auto product = a * b;
```

Comment:
Note that signed integer overflow is undefined behavior so we should make sure that `T` is unsigned if it's an integer type.

Reply:
Restricting this to unsigned integrals only.